### PR TITLE
Added support for cron expression

### PR DESCRIPTION
**Pull request by zinen**
I wanted to define some on and off time for this and are now testing with this code. 

Not fully tested yet. But are you interessed?

Added to docker-compose.yml with this PR:
```
      - SCHEDULE_CRON=                               ## (optional) Scheuler in cron schedule expression 
      ## Example of cron schedule expression:
      #¤ - SCHEDULE_CRON=*/1 12-14 * * 1-5     = allowed to run every 1 minute at hours 12-14 on monday to friday
      ## more help with formatting at https://crontab.guru/#*/1_12-14_*_*_1-5
```

### CODE REVIEW

File: src/scanner.py [hunk 1]
```diff
@@ -171,18 +173,25 @@ def run(self) -> NoReturn:
...
+                log.info("Schedule cron expression: " + get_description(self.config.schedule_cron))
             except Exception:
-                log.error("Job Error! - %s", sys.exc_info())
-            finally:
-                sleep(self.config.sleep_time * (0.9 + 0.2 * random()))
+                log.warning("Schedule cron expression parsing error - %s", sys.exc_info())
+                log.info("Schedule cron expression is ignored")
+                self.config.schedule_cron = '* * * * *'
+        while True:
+            if (pycron.is_now(self.config.schedule_cron)):
```
Comment (Der-Henning) [inline src/scanner.py:184]: Can this line raise an error?

**Comment by Der-Henning** [comment]
Hi @zinen.
Thank you for your effort. This seems to be a useful feature and I will be happy to include it. 
Please make sure that it is purely optional and it is possible to update without changing an existing config.
I haven't tested it yet but I will do so in the next days. I will create annotations if I see any issues.

File: src/models/config.py [hunk 2]
```diff
@@ -62,6 +63,7 @@ def _ini_reader(self, file: str) -> None:
         self.item_ids = config["MAIN"].get("ItemIDs").split(
             ',') if "ItemIDs" in config["MAIN"] else []
         self.sleep_time = config["MAIN"].getint("SleepTime")
+        self.schedule_cron = config["MAIN"].getint("ScheduleCron")
```
Comment (Der-Henning) [inline src/models/config.py:66]: This should be `.get("ScheduleCron")`, since it is a string?

File: src/scanner.py (hunk 1 above)
Comment (zinen) [reply src/scanner.py:184]: I'll have to look into the code of `pycron` to know for sure. The doc for the lib is rather thin. I'll get back to you on that.

**Comment by zinen** [comment]
> Hi @zinen. Thank you for your effort. This seems to be a useful feature and I will be happy to include it. Please make sure that it is purely optional and it is possible to update without changing an existing config. I haven't tested it yet but I will do so in the next days. I will create annotations if I see any issues.

Agreed. So far i should have input default settings make this setting both optional and so also seemless upgradeble for exsisting users without need to change config.

File: src/models/config.py (hunk 2 above)
Comment (zinen) [reply src/models/config.py]: Correct. Fixed now [78b408b](https://github.com/zinen/tgtg/commit/78b408b71bed3196735efd82a0c0b4866a375d1a)

File: src/scanner.py (hunk 1 above)
Comment (zinen) [reply src/scanner.py]: Looked into library `pycron` and `cron_descriptor` source code, Resulted in some changes arround multible places. [5cfeb2a](https://github.com/Der-Henning/tgtg/pull/142/commits/5cfeb2aa9b8d502bf4d599cb24b17cf8ffdd6755).
Testing this branch now via `docker pull zinen2/tgtg:edge-alpine``.

**Comment by zinen** [comment]
A few changes due to [5cfeb2a] should trigger a new review from you.

**Review by Der-Henning — COMMENTED** [review]
Please note these annotations.

File: src/scanner.py [hunk 3]
```diff
@@ -171,18 +173,26 @@ def run(self) -> NoReturn:
...
+                self.config.schedule_cron = None
         while True:
             try:
-                self._job()
-                if self.tgtg_client.captcha_error_count > 10:
-                    log.warning("Too many 403 Errors. Sleeping for 1 hour.")
-                    sleep(60 * 60)
-                    log.info("Continuing scanning.")
-                    self.tgtg_client.captcha_error_count = 0
+                if (not self.config.schedule_cron or pycron.is_now(self.config.schedule_cron)):
```
[the remark this answers is not in the document]
Comment (zinen) [reply src/scanner.py]: Done in [4a0da81](https://github.com/Der-Henning/tgtg/pull/142/commits/4a0da81aa3e4c54355c7c876268d5b12e488ec14)

File: src/scanner.py [hunk 4]
```diff
@@ -171,18 +173,26 @@ def run(self) -> NoReturn:
         Main Loop of the Scanner
         """
         log.info("Scanner started ...")
+        if (self.config.schedule_cron):
```
[the remark this answers is not in the document]
Comment (zinen) [reply src/scanner.py]: Cant put it inside `__init__` but added it to `main()`. Have to run after scanner init class is called. Look at [4a0da81](https://github.com/Der-Henning/tgtg/pull/142/commits/4a0da81aa3e4c54355c7c876268d5b12e488ec14)

**Comment by zinen** [comment]
Marked PR ready for merge

**Pull request merged**